api/app/signals/apps/signals/models/category_assignment.py

```diff
@@ -17,6 +18,16 @@ class CategoryAssignment(CreatedUpdatedModel):
 
     extra_properties = JSONField(null=True)  # TODO: candidate for removal
 
+    # Note on the deadline fields: nulls are allowed for historic data because
+    # historic service level objectives are not generally available.
+    deadline = models.DateTimeField(null=True)
+    deadline_factor_3 = models.DateTimeField(null=True)
+
     def __str__(self):
         """String representation."""
         return '{sub} - {signal}'.format(sub=self.category, signal=self._signal)
+
+    def save(self, *args, **kwargs):
```

Comment:
Talked about this, for now we leave this functionality in the `save` later this could/should be refactored to the desired implementation.

Maybe add a small comment explaining the code.

Reply:
comment added